pkg/webhooks/validation/controller/controller.go

```diff
@@ -241,14 +241,16 @@ func newController(
 }
 
 func (c *Controller) Run(stop <-chan struct{}) {
-	defer c.queue.ShutDown()
+	go func() {
```

Comment:
This is the actual fix. ValidationController is blocking and waiting for "stop". So the callback is never finished and `queue.done()` is not called and hence queue does not process the second secret.

Reply:
I think it is safer to modify https://github.com/istio/istio/blob/9b16ea834c7301808aaa1723e2237254ce791b7e/pilot/pkg/serviceregistry/kube/controller/multicluster.go#L259
run it in a separate goroutine

Reply:
>and hence queue does not process the second secret.


I am confused. The queue here is to handle ValidatingWebhookConfiguration

Reply:
Correct but it does not come out. It blocks the call back i.e. `AddMemberCluster`. It will never return.